### PR TITLE
Use more size_t in util/str

### PR DESCRIPTION
**Pull request by radare**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md#code-style)
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [radare2 book](https://github.com/radareorg/radare2book) with the relevant information (if needed)

**Detailed description**
size_t vs int 

**Test plan**
wait for travis

**Closing issues*

not an issue, just a standard to follow

### CODE REVIEW

File: libr/util/str.c [hunk 1]
```diff
@@ -1982,7 +1979,7 @@ R_API char *r_str_ansi_crop(const char *str, ut32 x, ut32 y, ut32 x2, ut32 y2) {
 	return ret;
 }
 
-R_API int r_str_utf8_codepoint (const char* s, int left) {
+R_API size_t r_str_utf8_codepoint(const char* s, size_t left) {
 	bool safe = left >= 0;
```
Comment (thestr4ng3r) [inline libr/util/str.c:1983]: Useless check, as reported by lgtm

**Comment by XVilka** [comment]
`size_t` limits the values to uint32, what happens if you meet 5Gb string from the file compiled on the 64bit computer but in radare2 compiled on 32bit one?

**Comment by karliss** [comment]
> `size_t` limits the values to uint32, what happens if you meet 5Gb string from the file compiled on the 64bit computer but in radare2 compiled on 32bit one?

In the situation you described `const char *` is also limited to 32 bit address space. All of these functions operate with strings that are already in memory. If you really wanted to operate with strings that don't fit into RAM you would need to write functions which accept position inside RIO address space instead of `char*`.

File: libr/util/str.c (hunk 1 above)
Comment (radare) [reply libr/util/str.c:1983]: fixed

**Pull request merged**